core/services/workflows/engine.go

```diff
@@ -680,6 +681,40 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 	}
 }
 
+func merge(baseConfig *values.Map, overrideConfig *values.Map) *values.Map {
+	m := values.EmptyMap()
+
+	for k, v := range baseConfig.Underlying {
+		m.Underlying[k] = v
+	}
+
+	for k, v := range overrideConfig.Underlying {
+		m.Underlying[k] = v
+	}
+
+	return m
+}
+
+func configForStep(ctx context.Context, reg core.CapabilitiesRegistry, localNode capabilities.Node, step *step) (*values.Map, error) {
+	ID := step.info.ID
+
+	// If the capability info is missing a DON, then
+	// the capability is local, and we should use the localNode's DON ID.
+	var donID uint32
+	if step.info.DON != nil && step.info.DON.ID > 0 {
```

Comment:
we are using the info.Don != nil in a couple of places (line 169, possibly other places) to determine if a capability is local, maybe add an isLocal method to CapabilityInfo to make this more obvious?

Reply:
Yes, but do you mind if I follow up with this change in a separate PR? This is getting quite large already and I'd like to keep changes to a minimum

Reply:
+1 to @ettec 

@cedric-cordenier if we don't do this now at minimum we need a TODO and a ticket in a comment

Reply:
I am going to address this, but in a slightly different way in this PR: the plan is to add a IsLocal flag (not function) and to set this explicitly. This is because @DeividasK gave me a good idea which should simplify this logic and just add the local DON info to local capabilities when fetching.